.travis.yml

```diff
@@ -38,18 +40,39 @@ before_install:
   - chmod +x docker-compose
   - sudo mv docker-compose /usr/local/bin
 
-before_script:
-  - echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin
+stages:
+- test
+- deploy
 
-script:
-  - docker-compose --file docker-compose-travis.yml build --build-arg BUILDKIT_INLINE_CACHE=1 
-  - docker-compose --file docker-compose-travis.yml up --exit-code-from todoapp
+jobs:
+  include:
+  - stage: test
+    script:
+    - echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin
+    - docker-compose --file docker-compose-travis.yml build --build-arg BUILDKIT_INLINE_CACHE=1 
+    - docker-compose --file docker-compose-travis.yml up --exit-code-from todoapp
 
-deploy:
-  - provider: script
-    script: bash docker_push
-    on:
-      branch: main
+  - stage: deploy
+    if: branch = main
```

Comment:
Note that this will be true for pull requests into `main`, as pull request builds use the base branch for this variable, see <https://docs.travis-ci.com/user/conditional-builds-stages-jobs/#conditional-builds>

Reply:
I did get confused by this, but had run the terraform deployment in previous travis build before the PR.  I've removed the if and will try the on statement later